### PR TITLE
Update haproxy-best-practices.md

### PR DESCRIPTION
**Pull request by Chad20N13**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/7114
- Other reference link(s): https://github.com/pingcap/docs/pull/7046

### Do your changes match any of the following descriptions?

- [NO] Delete files
- [NO] Change aliases
- [NO] Need modification after applied to another branch 
- [NO] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -168,7 +168,7 @@ global                                     # 全局配置。
    log         127.0.0.1 local2            # 定义全局的 syslog 服务器，最多可以定义两个。
    chroot      /var/lib/haproxy            # 更改当前目录并为启动进程设置超级用户权限，从而提高安全性。
    pidfile     /var/run/haproxy.pid        # 将 HAProxy 进程的 PID 写入 pidfile。
-   maxconn     256                         # 每个 HAProxy 线程可接受的最大并发连接数。
+   maxconn     4096                        # 单个HAProxy进程可接受的最大并发连接数。
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:171]:
```suggestion
   maxconn     4096                        # 单个 HAProxy 进程可接受的最大并发连接数。
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Comment by lilin90** [comment]
@superlzs0476 PTAL

**Review by superlzs0476 — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ global                                     # 全局配置。
    log         127.0.0.1 local2            # 定义全局的 syslog 服务器，最多可以定义两个。
    chroot      /var/lib/haproxy            # 更改当前目录并为启动进程设置超级用户权限，从而提高安全性。
    pidfile     /var/run/haproxy.pid        # 将 HAProxy 进程的 PID 写入 pidfile。
-   maxconn     256                         # 每个 HAProxy 线程可接受的最大并发连接数。
+   maxconn     4096                        # 单个 HAProxy 进程可接受的最大并发连接数。（ 等价于命令行参数 "-n" )
```
Comment (TomShawn) [inline best-practices/haproxy-best-practices.md:171]:
```suggestion
   maxconn     4096                        # 单个 HAProxy 进程可接受的最大并发连接数，等价于命令行参数 "-n"。
```

**Pull request merged**